scripts/py_matter_idl/files.gni

```diff
@@ -32,29 +37,39 @@ matter_idl_generator_sources = [
   "${chip_root}/scripts/py_matter_idl/matter/idl/generators/cpp/tlvmeta/__init__.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/generators/filters.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/generators/generator.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/generators/idl/__init__.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/generators/java/__init__.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/generators/registry.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/generators/type_definitions.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/lint/__init__.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/lint/lint_rules_parser.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/lint/type_definitions.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/matter_idl_parser.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/matter_idl_types.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/test_backwards_compatibility.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/test_data_model_xml.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/test_generators.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/test_matter_idl_parser.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/test_zapxml.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/__init__.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/handlers/__init__.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/handlers/base.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/handlers/context.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/handlers/handlers.py",
   "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml/handlers/parsing.py",
-  "${chip_root}/scripts/py_matter_idl/matter/idl/zapxml_parser.py",
+  "${chip_root}/scripts/py_matter_idl/scripts/matter-data-model-xml-parser.py",
+  "${chip_root}/scripts/py_matter_idl/scripts/matter-idl-lint-parser.py",
+  "${chip_root}/scripts/py_matter_idl/scripts/matter-idl-parser.py",
+  "${chip_root}/scripts/py_matter_idl/scripts/matter-zapxml-parser.py",
+]
+
+matter_idl_generator_tests = [
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_backwards_compatibility.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_case_conversion.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_data_model_xml.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_matter_idl_parser.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_generators.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_idl_generator.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_supported_types.py",
+  "${chip_root}/scripts/py_matter_idl/matter/idl/test_zapxml.py",
 ]
 
 # All the files that the matter idl infrastructure will use
 matter_idl_generator_files =
-    matter_idl_generator_templates + matter_idl_generator_sources
+    matter_idl_generator_templates + matter_idl_generator_sources +
```

Comment:
These were added here to be shared with codegen_data_model (i.e. ask for model regeneration whenever the underlying generator errors out).

However this does not make sense for tests: there is no reason to trigger a regeneration when a test changes. Did we have a reason of placing `matter_idl_generator_tests` here other than consistency? I would rather move it back in its only use place if there is no other reason.

Reply:
Yes, I've pulled it here for consistency only. I will move it back.